### ISSUE TITLE
Commonmark compliant lists

### ISSUE THREAD

**Issue by joncass** [miyuchina/mistletoe#40]
See [this example](https://spec.commonmark.org/dingus/?text=1.%20List%20item%201%0A%20%20%20%20%60%60%60%0A%20%20%20%20This%20is%20code%0A%20%20%20%20%60%60%60%0A1.%20This%20should%20be%20numbered%20as%202%0A%0A%20%20%20%20This%20is%20indented%0A%0A1.%20Number%203)

In Mistletoe the indentations/codeblocks get mixed up, and the numbering in the list does not continue correctly

**Comment by miyuchina** [comment]
I think I know how to fix this one, but it'll take me a bit. Lists are hard to parse, and I'll have to think of a better way.

**Comment by joncass** [comment]
Sure, makes sense. When you say "a bit" are you thinking days, weeks, months, ...? No pressure - I just want to get an idea of whether I need to consider a workaround or if I can wait.

**Comment by miyuchina** [comment]
Almost there. Some whitespace still need fixing.

I also broke backwards compatibility (re: `BlockToken.start`) and need to write new test cases for the new list implementation, so the next release will be delayed for ~2 days, but hopefully not more than that.

**Comment by joncass** [comment]
Very cool - thanks for the update and continued responsiveness!

**Comment by miyuchina** [comment]
I just released a new version. Output for the original example:

```html
```

**Comment by joncass** [comment]
I pulled the new release into my project, and while it does fix the exact example pasted above, it unfortunately doesn't work for this variation (note the extra newline before the codeblock starts):
````
mistletoe [version 0.6] (interactive)
Type Ctrl-D to complete input, or Ctrl-C to exit.
>>> 1. List item 1
...
...     ```
...     This is code
...     ```
... 1. This is number 2
...
<ol>
<li><p>List item 1</p>
<p><code>`</code>
This is code</p>
<pre><code></code></pre>
</li>
<li>This is number 2</li>
</ol>
>>>
````

I also noticed that it adds spurious instances of `<pre><code></code></pre>` when there are lines that only consist of whitespace. Probably not a huge deal for most cases, but it seemed odd to me.

Let me know what you think about the above issue(s)! Thanks!

**Comment by miyuchina** [comment]
I pushed a temporary fix, though it has less to do with parsing lists, and more to do with how code fences are currently parsed. I'll open another issue for implementing Commonmark-compliant code fences. Currently the example is working:

```html
>>> 1. List item 1
...
...     ```
...     This is code
...     ```
... 1. This should be numbered as 2
...
<ol>
<li><p>List item 1</p>
<pre><code>This is code
</code></pre>
</li>
<li>This should be numbered as 2</li>
</ol>
```

Can you give an example of spurious `<pre><code></code></pre>`?
